versioned/storage/telemetry/src/main/java/org/projectnessie/versioned/storage/telemetry/TelemetryPersist.java

```diff
@@ -0,0 +1,405 @@
+      } catch (ObjTooLargeException e) {
+        trace.attribute("error", "too large");
+        throw e;
+      } catch (RuntimeException e) {
+        throw trace.unhandledError(e);
+      }
+    }
+  }
+
+  @Override
+  @Nonnull
+  @jakarta.annotation.Nonnull
+  public boolean[] storeObjs(@Nonnull @jakarta.annotation.Nonnull Obj[] objs)
+      throws ObjTooLargeException {
+    try (Traced trace = traced("storeObjs").attribute("objs.length", objs.length)) {
+      stream(objs)
+          .collect(groupingBy(Obj::type, counting()))
+          .forEach((t, c) -> trace.attribute("type." + t.name() + ".count", c));
+      try {
+        boolean[] result = persist.storeObjs(objs);
+        int successes = 0;
+        for (boolean b : result) {
+          if (b) {
+            successes++;
+          }
+        }
+        trace.attribute("result.nonNull", successes);
+        return result;
+      } catch (ObjTooLargeException e) {
+        trace.attribute("error", "too large");
+        throw e;
+      } catch (RuntimeException e) {
+        throw trace.unhandledError(e);
+      }
+    }
+  }
+
+  @Override
+  public void deleteObj(@Nonnull @jakarta.annotation.Nonnull ObjId id) {
+    try (Traced trace = traced("deleteObj")) {
+      try {
+        persist.deleteObj(id);
+      } catch (RuntimeException e) {
+        throw trace.unhandledError(e);
+      }
+    }
+  }
+
+  @Override
+  public void deleteObjs(@Nonnull @jakarta.annotation.Nonnull ObjId[] ids) {
+    try (Traced trace = traced("deleteObjs").attribute("ids.length", ids.length)) {
+      try {
+        persist.deleteObjs(ids);
+      } catch (RuntimeException e) {
+        throw trace.unhandledError(e);
+      }
+    }
+  }
+
+  @Override
+  public void updateObj(@Nonnull @jakarta.annotation.Nonnull Obj obj)
+      throws ObjTooLargeException, ObjNotFoundException {
+    try (Traced trace = traced("updateObj")) {
+      try {
+        persist.updateObj(obj);
+      } catch (RuntimeException e) {
+        throw trace.unhandledError(e);
+      }
+    }
+  }
+
+  @Override
+  public void updateObjs(@Nonnull @jakarta.annotation.Nonnull Obj[] objs)
+      throws ObjTooLargeException, ObjNotFoundException {
+    try (Traced trace = traced("updateObjs")) {
+      try {
+        persist.updateObjs(objs);
+      } catch (RuntimeException e) {
+        throw trace.unhandledError(e);
+      }
+    }
+  }
+
+  @Override
+  @Nonnull
+  @jakarta.annotation.Nonnull
+  public CloseableIterator<Obj> scanAllObjects(
+      @Nonnull @jakarta.annotation.Nonnull Set<ObjType> returnedObjTypes) {
+    Traced trace = traced("scanAllObjects");
+    boolean ok = false;
+    try {
+      @SuppressWarnings("resource")
+      CloseableIterator<Obj> iter = persist.scanAllObjects(returnedObjTypes);
+      ok = true;
+      return new CloseableIterator<Obj>() {
+        @Override
+        public void close() {
+          try {
+            iter.close();
+          } finally {
+            trace.close();
```

Comment:
I think this is not correct. Since the `span` is made "current" we should not return from this method without closing the current `scope`. The caller may close its `scope` before us and that would be a violation of OTel spec, AFAIK.

Reply:
In this case I'd propose closing the `scope` upon returning from this method, but closing / ending the `span` when the stream is exhausted. This way any nested spans will be properly linked to this `span`, while the caller will be able to exit or start sibling spans correctly. Out `span` will become "detached" (may stretch longer than the parent span). To be strictly correct in this situation I think we ought to have a pair or "producer" and "consumer" spans - the producer would close on return, the consumer would track the lifetime of the stream... but even with one "non-current" span it might be good enough.

Reply:
An even more correct approach would be to use two spans and activate/deactivate the inner (stream) span of every stream methods... but that would be an overkill... Do we really need to track the stream's lifetime as a span?

Reply:
Hm - it's a rather maintenance-only operation, so probably not worth to add a ton of code here.